src/backend/opencl/kernel/copy.cl

```diff
@@ -8,16 +8,14 @@
  ********************************************************/
 
 typedef struct {
-    dim_t dim[4];
+    int dims[4];
```

Comment:
In this case, I change the struct name to dims_type.

Reply:
is there a reason we can't use `KParam` like the other kernels ?

Reply:
I already had all the dims available in int format and no longer in dim_t
format, because they are updated inside the 'memcopy.hpp calls'.
On top, most of the OpenCL kernels are converting all dims / idx into int
inside their kernels.

Finally, I wanted to speed things up more, avoiding conversions and
reducing the bytes sent to the GPU.
If I have to send the dim_t version, I have to convert on the CPU and later
convert again on GPU.

Continuing calculations in dim_t format is really much slower than int on
OpenCL.  I did not have the same impression on CUDA, even when running on
the same HW.

Reply:
I understand the performance implications of using `dim_t` vs `int`. In fact, we originally had `dim_t` all over but later refactored to current state where we keep `dim_t` outside of kernels but convert them to `int` inside kernels for the very same reason. This is case with even CUDA backend, not just OpenCL. Type conversion overhead for a few scalars shouldn't be that noticeable.

I don't understand why you would have to convert from int to dim_t on the host. Can you please point me to the source location where these are happening. As far as I can remember, all non-kernel codes almost always deals in terms of `dim_t` and `dim_t` to `int` inside kernels alone.

Reply:
I updated dims_t to dims_type

Reply:
I was referring to memcopy.hpp, since the dims and strides are updated in array dimension optimizations.  This is all done in int, since the kernel continuous in int.

So the latest values are only available in int in opencl/memcopy.hpp